src/test/login-test.ts

```diff
@@ -0,0 +1,151 @@
+import axios, { AxiosResponse } from 'axios';
+import bcrypt from 'bcrypt';
+import { expect } from 'chai';
+import { serverUrl } from '../setup-server';
+import { AppDataSource } from '../data-source.js';
+import { User } from '../entity/User.js';
+
+interface CreateUserInputData {
+  name: string;
+  email: string;
+  birthDate: string;
+  password: string;
+}
+
+interface LoginInputData {
+  email: string;
+  password: string;
+}
+
+async function postLogin(loginInputData: LoginInputData): Promise<AxiosResponse> {
+  const response = await axios.post(`${serverUrl}graphql`, {
+    query: `
+    mutation Login($data: LoginInput!) {
+      login(data: $data) {
+        user {
+          birthDate
+          email
+          id
+          name
+        }
+        token
+      }
+    }
+      `,
+    variables: {
+      data: loginInputData,
+    },
+  });
+
+  return response;
+}
+
+describe('login mutation', () => {
+  beforeEach(async () => {
+    const userRepository = AppDataSource.getRepository(User);
+    await userRepository.clear();
+  });
+
+  it('should be able to login', async () => {
+    const inputUser: CreateUserInputData = {
+      name: 'John Phill',
+      email: 'john@example.com',
+      password: 'password123',
+      birthDate: '1990-01-01',
+    };
+
+    const inpuLogin: LoginInputData = {
+      email: inputUser.email,
+      password: inputUser.password,
+    };
+
+    const inputUserWithoutPassword = {
+      name: inputUser.name,
+      email: inputUser.email,
+      birthDate: inputUser.birthDate,
+    };
+
+    const expectedResponseWithoutPassword = {
+      user: inputUserWithoutPassword,
+      token: '',
+    };
+
+    const userRepository = AppDataSource.getRepository(User);
```

Comment:
Vc poderia ter o userRepository numa variável "global" do escopo desse teste (que é declarada no describe). Claro que essa prática de variável compartilhada entre os its tem que ser usada com cuidado, mas nesse caso que é só o repository, tudo bem:


```ts
describe('Uma query/mutation', () => {
  let repository: Repository<User>;

  before(() => {
    userRepository = AppDataSource.getRepository(User);
  });
})
```

Agora todos os lugares que vc precisar fazer alguma operação no banco com usuário, vc tem o repository pronto na variável.

Reply:
Ótima ideia! Só não entendi a necessidade do `let repository: Repository<User>;`

Reply:
Eu tentei  implementar isso, mas essa declaração feita no `before` parece não estar sendo compartilhada com o `beforeEach` e entre os `its`

Reply:
Então vc já tem a resposta porque precisa do `let repository: Repository<User>;` 🤣 

É que pras funções `beforeEach` e `it` terem acesso a uma variável compartilhada, ela precisa ser declarada no escopo do `describe`, que é um escopo acima dessas funções, correto? Qualquer coisa dá uma lida sobre escopos.

Reply:
Aaah, faz todo sentido sim